### ISSUE TITLE
[1.16.1 Preview] Piglins spawned in bastion appear invisible and cannot be attacked.

### ISSUE THREAD

**Issue by WinterLift** [sp614x/optifine#4322]
I've played with optifine and piglins spawned in other areas do not have any problems. However when I entered a bastion I was unable to see any Piglins. They hit me multiple times and I was also unable to hit the piglins attacking me. I saved and quit the game then reloaded on normal 1.16.1 Minecraft. The piglins appeared perfectly fine and could be damaged.

Update: It appears Piglins not spawned in a bastion can also do this. Leaving and rejoining seems to fix this problem.

**Comment by thyIacoIeo** [comment]
I am also experiencing this(both on Optifine 1.16 pre2 and pre4).

It appears that any hostile mob nbtagged as Persistent has the potential to become invisible. This includes nametagged mobs and structure-spawning mobs such as Woodland Mansion Vindicators + Bastion Piglins/Hoglins.

I can reproduce this by spawning hostile mobs, nametagging them, and leaving render distance. After switching between shaders and returning to render distance, some or all of the mobs will be invisible. To make them visible again, repeat steps.

I am unsure if passive mobs can turn invisible since an invisible trout won’t be noticeable.

**Comment by rbits0** [comment]
I'm experiencing this as well. It has happened to me many times with skeletons and creepers that weren't nametagged, just naturally spawning, so I think it's also possible for mobs that aren't persistent to turn invisible as well.

**Comment by johnvilsack** [comment]
FIX: Disabling Dynamic Lighting seems to solve the problem.

This is not ideal, considering it's the most valuable aspect of Optifine for me.

**Comment by sp614x** [comment]
Seems to be fixed with 1.16.2 pre6.

**Comment by mcdj23** [comment]
bug still seems to be there and is really annoying and I'm running the new release of optifine I will post a video next time it happens also restart mc seems to temp fix it

**Comment by mcdj23** [comment]
test 1 was of my modded client using fabric with optifabric and optifine and a few other client-side mods
https://youtu.be/bMQ-0Dvc_Rw 

test 2 was of vanilla with optifine https://youtu.be/HNHY65Frk-k

as you can see it happens on both also this was all done on a realms server
if any more info is needed you can msg me on discord mcdj23#4292

**Comment by willtheoct** [comment]
Would you be able to switch the 'latest release' on the website to have the pre6 version? it seems fixed in the preview but the default download which most people will get has this bug, and this bug makes the game a little bit unplayable

**Comment by BrandonPotter0** [comment]
The invisible/invulnerable mob issue is still an issue even today. There has been no fix yet.

**Comment by willtheoct** [comment]
> 
> 
> The invisible/invulnerable mob issue is still an issue even today. There has been no fix yet.

fixed for me in g4_pre2 ultra

**Comment by BrandonPotter0** [comment]
> > The invisible/invulnerable mob issue is still an issue even today. There has been no fix yet.
> 
> fixed for me in g4_pre2 ultra

I actually have the issue in G4 pre2, so yeah, not fixed yet.

**Comment by ksprwhite** [comment]
still happening in 1.16.3 G4 pre2

**Comment by chasejsutton49** [comment]
Confirmed for 1.16.4 Optifine HD U G5
Can confirm that this is not a vanilla bug.

**Comment by CKaylis** [comment]
Just happened for me as well today, same 1.16.4 HD U G5 version.

**Comment by JamesTheAwesomeDude** [comment]
This might not be an OptiFine bug.

My brother and I are both affected, but he's using an entirely vanilla client.

**Comment by chasejsutton49** [comment]
*Potentially* fixed in 1.16.4 Optifine HD U G6 pre12, but don't quote me on that.
Went into a bastion remnant where I was experiencing this bug, Piglins have spawned inside.

**Comment by SkyeMors** [comment]
Encountering this (invisible hostile mobs) in MC 1.16.5 with OptiFine_1.16.5_HD_U_G6 and Forge 36.0.0.
Walking around, hear Hmm, assume it's a traveling merchant, get shot at, assume it was a skeleton, nothing nearby and Hmming getting louder, seems there's 2-3 wandering Pillagers that are active but is otherwise unloaded. No model, no shadow, still show up on Xaero's minimap. Reloading world (singleplayer) reloads everything, including the enemies, and therefor causing the pillagers to load in fully.

Basically: Found in MC1.16.5 w/ Pillagers in Overworld, server reload fixes (temporarily?)

**Comment by Erdogan-0** [comment]
I had the same issue as SkyeMors on OptiFine HD U G6
I was in a bastion and heard a bunch of piglins. I dropped a gold ingot down into the area where I heard piglin noises and saw it get picked up though no piglin was in sight. I dropped down into the pit and got ambushed by an invisible piglin brute.

**Comment by NoMoreStars** [comment]
I was also attacked by an invisible zombie villager the other day, as well as encountering villagers with missing textures, making them see-through on parts of the model.
